### PR TITLE
add docs for v5.1: reference -> architecture

### PR DESCRIPTION
**Pull request by marsishandsome**
### What is changed?

TiKV's official website is planned to be refactored as discussed in https://github.com/tikv/website/issues/227.

This PR covers `Reference` -> `Architecture`

### Any related PRs or issues?

https://github.com/tikv/website/issues/227.

### Which version does your change affect?
v5.1

### CODE REVIEW

**Comment by marsishandsome** [comment]
@andylokandy @zz-jason @iosmanthus PTAL

File: content/docs/5.1/reference/architecture/overview.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        parent: Architecture
+        weight: 1
+---
+
+This page discusses the core concepts and architecture behind TiKV, including:
+
+* The [APIs](#apis) that applications can use to interact with TiKV
+* The basic [system architecture](#system) underlying TiKV
+* The anatomy of each [instance](#instance) in a TiKV installation
+* The role of core system components, including the [Placement Driver](#placement-driver), [Store](#store), [Region](#region), and [Node](#node)
```
Comment (iosmanthus) [inline content/docs/5.1/reference/architecture/overview.md:15]: Do you mean `the terminology of the TiKV cluster`

Comment (iosmanthus) [reply content/docs/5.1/reference/architecture/overview.md:15]: Since, region is not a "role", it's a terminology of our own.

Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md:15]: yes

Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md]: fixed

File: content/docs/5.1/reference/architecture/overview.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+* The role of the [Raft consensus algorithm](#raft) in TiKV
+* The [origins](#origins) of TiKV
+
+## APIs
+
+TiKV provides two APIs that you can use to interact with it:
+
+| API           | Description                                                                           | Atomicity     | Use when...                                                                                                            |
+|:------------- |:------------------------------------------------------------------------------------- |:------------- |:---------------------------------------------------------------------------------------------------------------------- |
+| Raw           | A lower-level key-value API for interacting directly with individual key-value pairs. | Single key    | Your application requires low latency instead of distributed transactions or multi-version concurrency control (MVCC). |
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:26]:
```suggestion
| Raw           | A lower-level key-value API for interacting directly with individual key-value pairs. | Single key    | Your application requires low latency and doesn't use multi-key transactions. |
```

File: content/docs/5.1/reference/architecture/overview.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+| Transactional | A higher-level key-value API that provides ACID semantics                             | Multiple keys | Your application requires distributed transactions and/or MVCC.                                                        |
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:27]:
```suggestion
| Transactional | A higher-level key-value API that provides snapshot isolation transaction                             | Multiple keys | Your application requires distributed transactions.                                                        |
```

File: content/docs/5.1/reference/architecture/overview.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## Placement driver (PD) {#placement-driver}
+
+The TiKV placement driver is the cluster manager of TiKV, which periodically checks replication constraints to balance load and data automatically across nodes and regions in a process called **auto-sharding**.
+
+## Store
+
+There is a [RocksDB](https://rocksdb.org) database within each **Store** and it stores data into the local disk.
+
+## Region
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:58]: May be better to explain peer after this.

File: content/docs/5.1/reference/architecture/overview.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+**Region** is the basic unit of key-value data movement. Each Region is replicated to multiple Nodes. These multiple replicas form a Raft group.
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:60]:
```suggestion
TiKV shards continuous ranges of keys into **Regions**, and replicates **Regions** via the Raft protocol. When data size increases until reaching a threshold, a Region will be split into multiple. Conversely, if the size of the Region shrinks due to data deletion, two adjacent Regions can be merged into one.
```

File: content/docs/5.1/reference/architecture/overview.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    width="60" >}}
+
+
+## Placement driver (PD) {#placement-driver}
+
+The TiKV placement driver is the cluster manager of TiKV, which periodically checks replication constraints to balance load and data automatically across nodes and regions in a process called **auto-sharding**.
+
+## Store
+
+There is a [RocksDB](https://rocksdb.org) database within each **Store** and it stores data into the local disk.
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:56]:
```suggestion
The **Store** is a TiKV instance. A Store stores multiple peers.
```

move below peer section.

File: content/docs/5.1/reference/architecture/overview.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+## Region
+
+**Region** is the basic unit of key-value data movement. Each Region is replicated to multiple Nodes. These multiple replicas form a Raft group.
+
+## Node
+
+A TiKV **Node** is just a physical node in the cluster, which could be a virtual machine, a container, etc. Within each Node, there are one or more stores. The data in each Store is split across multiple regions. Data is distributed across Regions using the Raft algorithm.
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:64]:
```suggestion
A TiKV **Store** is a physical node in the cluster, which could be a virtual machine, a container, etc. Within each Node, there can be one or more stores.
```

File: content/docs/5.1/reference/architecture/overview.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+When a Node starts, the metadata for the Node, Store, and Region is recorded into the Placement Driver. The status of each Region and Store is regularly reported to the PD.
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:66]:
```suggestion
The status for the Node, Store, and Region will be regularly reported to the Placement Driver.
```

File: content/docs/5.1/reference/architecture/overview.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+## Transaction model {#transactions}
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:68]:
```suggestion
## Transaction {#transactions}
```

File: content/docs/5.1/reference/architecture/overview.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+{{< figure
+    src="/img/tikv-instance.png"
+    caption="TiKV instance architecture"
+    alt="TiKV instance architecture diagram"
+    width="60" >}}
+
+
+## Placement driver (PD) {#placement-driver}
+
+The TiKV placement driver is the cluster manager of TiKV, which periodically checks replication constraints to balance load and data automatically across nodes and regions in a process called **auto-sharding**.
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:52]:
```suggestion
The TiKV placement driver is the cluster manager of TiKV, which periodically and automatically balances workload and data storage across nodes by moving Regions. This process is called **auto-sharding**.
```

File: content/docs/5.1/reference/architecture/overview.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+When a Node starts, the metadata for the Node, Store, and Region is recorded into the Placement Driver. The status of each Region and Store is regularly reported to the PD.
+
+## Transaction model {#transactions}
+
+TiKV's transaction model is similar to that of Google's [Percolator](https://ai.google/research/pubs/pub36726), a system built for processing updates to large data sets. Percolator uses an incremental update model in place of a batch-based model.
+
+TiKV's transaction model provides:
+
+* **Snapshot isolation** with lock, with semantics analogous to `SELECT ... FOR UPDATE` in SQL
+* Externally consistent reads and writes in distributed transactions
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:75]:
```suggestion
TiKV provides transactions that allow you to read and write across any keys with **Snapshot Isolation** regardless of the physical placement of the Region. TiKV also provides the pessimistic transactions that are semantically analogous to `SELECT ... FOR UPDATE` in SQL.

The underlying transaction model is similar to Google's [Percolator](https://ai.google/research/pubs/pub36726), a system built for processing updates to large data sets.
```

File: content/docs/5.1/reference/architecture/overview.md (hunk 4 above)
Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md:58]:
```
## Peer

A replica of a Region is called a peer. Multiple peers of the same Region replicate data via the Raft consensus algorithm, so peers are also members of a Raft instance.
```

File: content/docs/5.1/reference/architecture/overview.md (hunk 6 above)
Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md]: what about

```
A **Store** refers to the storage node in the TiKV cluster (an instance of tikv-server). Each store has a corresponding TiKV instance.
```

**Comment by marsishandsome** [comment]
@andylokandy PTAL

File: content/docs/5.1/reference/architecture/overview.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+    width="60" >}}
+
+
+## Placement driver (PD)
+
+The TiKV placement driver is the cluster manager of TiKV, which periodically and automatically balances workload and data storage across nodes by moving Regions. This process is called **auto-sharding**.
+
+## Store
+
+A **Store** refers to the storage node in the TiKV cluster (an instance of tikv-server). Each store has a corresponding TiKV instance.
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:56]: This is ambiguous to the `Node` section: Is node a tikv instance or the machine that runs the tikv?

File: content/docs/5.1/reference/architecture/overview.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+## Region
+
+TiKV shards continuous ranges of keys into **Regions**, and replicates **Regions** via the Raft protocol. When data size increases until reaching a threshold, a Region will be split into multiple. Conversely, if the size of the Region shrinks due to data deletion, two adjacent Regions can be merged into one.
+
+## Peer
+
+A replica of a Region is called a peer. Multiple peers of the same Region replicate data via the Raft consensus algorithm, so peers are also members of a Raft instance.
```
Comment (andylokandy) [inline content/docs/5.1/reference/architecture/overview.md:64]:
```suggestion
A replica of a Region is called a peer. Multiple peers of the same Region replicate data via the Raft consensus algorithm.
```

**Review by andylokandy — APPROVED** [review]
rest LGTM

File: content/docs/5.1/reference/architecture/overview.md (hunk 12 above)
Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md]: what about 
```
A **Store** is an instance of tikv-server.
```

File: content/docs/5.1/reference/architecture/storage.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+## Key-Value pairs
+
+The first thing to decide for a data storage system is the data storage model, that is, in what form the data is saved. TiKV's choice is the Key-Value model and provides an ordered traversal method. There are two key points for TiKV data storage model:
+
++ This is a huge Map (similar to `std::Map` in C++) , which stores Key-Value pairs.
++ The Key-Value pair in the Map is ordered according to Keys' binary order, which means you can Seek the position of a particular Key and then call the Next method to get the Key-Value pairs larger than this Key in incremental order.
+
+## Local storage (RocksDB)
+
+For any persistent storage engine, data is eventually saved on disk, and TiKV is no exception. TiKV does not write data directly on the disk, but stores data in RocksDB, which is responsible for the data storage. The reason is that it costs a lot to develop a standalone storage engine, especially a high-performance standalone engine that requires careful optimization.
```
[the remark this answers is not in the document]
Comment (zz-jason) [reply content/docs/5.1/reference/architecture/storage.md:21]: no need to explain the technical reasons for users.

File: content/docs/5.1/reference/architecture/overview.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+TiKV shards continuous ranges of keys into **Regions**, and replicates **Regions** via the Raft protocol. When data size increases until reaching a threshold, a Region will be split into multiple. Conversely, if the size of the Region shrinks due to data deletion, two adjacent Regions can be merged into one.
+
+## Peer
+
+A replica of a Region is called a peer. Multiple peers of the same Region replicate data via the Raft consensus algorithm.
+
+## Node
+
+A TiKV **Node** is a physical node in the cluster, which could be a virtual machine, a container, etc. Within each Node, there can be one or more stores.
```
[the remark this answers is not in the document]
Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md:68]: No, Store is the tikv-server process.

File: content/docs/5.1/reference/architecture/overview.md (hunk 12 above)
Comment (andylokandy) [reply content/docs/5.1/reference/architecture/overview.md]: Modified a little bit:

A **Store** is an instance of a TiKV server in which multiple peers are stored.

File: content/docs/5.1/reference/architecture/overview.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+{{< figure
+    src="/img/tikv-instance.png"
+    caption="TiKV instance architecture"
+    alt="TiKV instance architecture diagram"
+    width="60" >}}
+
+
+## Placement driver (PD)
+
+The TiKV placement driver is the cluster manager of TiKV, which periodically and automatically balances workload and data storage across nodes by moving Regions. This process is called **auto-sharding**.
```
[the remark this answers is not in the document]
Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md]: +1

File: content/docs/5.1/reference/architecture/overview.md [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+
+This page discusses the core concepts and architecture behind TiKV, including:
+
+* The [APIs](#apis) that applications can use to interact with TiKV
+* The basic [system architecture](#system) underlying TiKV
+* The anatomy of each [instance](#instance) in a TiKV installation
+* TiKV's terminology including [Placement Driver](#placement-driver), [Store](#store), [Region](#region), and [Node](#node)
+* TiKV's [transaction model](#transactions)
+* The role of the [Raft consensus algorithm](#raft) in TiKV
+* The [origins](#origins) of TiKV
```
[the remark this answers is not in the document]
Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md]: +1

[the remark this answers is not in the document]
Comment (marsishandsome) [reply content/docs/5.1/reference/architecture/overview.md]: PTAL

File: content/docs/5.1/reference/architecture/overview.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+
+TiKV Clients interact with PD and TiKV through gRPC.
+
+## TiKV Cluster
+
+TiKV stores data in RocksDB, which is a persistent key-value store for fast storage environment. This [article](/deep-dive/key-value-engine/rocksdb/) explains why RocksDB is selected.
+
+TiKV replicates data to multiple machines via [Raft](/deep-dive/consensus-algorithm/raft/) in case of machine failure. Data is written through the interface of Raft instead of to RocksDB. With the implementation of Raft, TiKV becomes a distributed Key-Value storage. Even with a few machine failures, TiKV can automatically complete replicas by virtue of the native Raft protocol, which does not impact the application.
+
+Based on the Raft layer, TiKV provides two APIs that clients can use to interact with it:
```
Comment (zz-jason) [inline content/docs/5.1/reference/architecture/overview.md:34]:
```suggestion
Based on the Raft layer, TiKV provides two APIs that clients can interact with:
```

**Pull request merged**